primitives/block/src/equivocation_proof.rs

```diff
@@ -326,21 +326,21 @@ pub struct DoubleProposalProof {
 impl DoubleProposalProof {
     pub fn new(
         validator_address: Address,
-        mut header1: MacroHeader,
+        mut proposal1: TendermintProposal<MacroHeader>,
         mut justification1: SchnorrSignature,
-        mut header2: MacroHeader,
+        mut proposal2: TendermintProposal<MacroHeader>,
         mut justification2: SchnorrSignature,
     ) -> DoubleProposalProof {
-        let hash1: Blake2bHash = header1.hash();
-        let hash2: Blake2bHash = header2.hash();
+        let hash1: Blake2bHash = proposal1.proposal.hash();
+        let hash2: Blake2bHash = proposal2.proposal.hash();
```

Comment:
Not done for now because we currently don't do that for the creation of any equivocation proof structure. We could automatically call `verify` I guess and `assert!` that it verifies. But maybe we could also do that at the call sites?